pkg/apiserver/rest/asset_scan_estimation_controller.go

```diff
@@ -0,0 +1,151 @@
+	if err != nil {
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan estimations results from db: %v", err))
+	}
+
+	return sendResponse(ctx, http.StatusOK, dbAssetScanEstimations)
+}
+
+func (s *ServerImpl) PostAssetScanEstimations(ctx echo.Context) error {
+	var assetScanEstimation models.AssetScanEstimation
+	err := ctx.Bind(&assetScanEstimation)
+	if err != nil {
+		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
+	}
+
+	createdAssetScanEstimation, err := s.dbHandler.AssetScanEstimationsTable().CreateAssetScanEstimation(assetScanEstimation)
+	if err != nil {
+		var conflictErr *common.ConflictError
+		if errors.As(err, &conflictErr) {
+			existResponse := &models.AssetScanEstimationExists{
+				Message:             utils.PointerTo(conflictErr.Reason),
+				AssetScanEstimation: &createdAssetScanEstimation,
+			}
+			return sendResponse(ctx, http.StatusConflict, existResponse)
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to create asset scan estimation in db: %v", err))
+	}
+
+	return sendResponse(ctx, http.StatusCreated, createdAssetScanEstimation)
+}
+
+func (s *ServerImpl) GetAssetScanEstimationsAssetScanEstimationID(ctx echo.Context, assetScanEstimationID models.AssetScanEstimationID, params models.GetAssetScanEstimationsAssetScanEstimationIDParams) error {
+	dbAssetScanEstimation, err := s.dbHandler.AssetScanEstimationsTable().GetAssetScanEstimation(assetScanEstimationID, params)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return sendError(ctx, http.StatusNotFound, err.Error())
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan estimation from db. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+	}
+
+	return sendResponse(ctx, http.StatusOK, dbAssetScanEstimation)
+}
+func (s *ServerImpl) PatchAssetScanEstimationsAssetScanEstimationID(ctx echo.Context, assetScanEstimationID models.AssetScanEstimationID, params models.PatchAssetScanEstimationsAssetScanEstimationIDParams) error {
+	var assetScanEstimation models.AssetScanEstimation
+	err := ctx.Bind(&assetScanEstimation)
+	if err != nil {
+		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
+	}
+
+	// check that an asset scan estimation with that id exists.
+	_, err = s.dbHandler.AssetScanEstimationsTable().GetAssetScanEstimation(assetScanEstimationID, models.GetAssetScanEstimationsAssetScanEstimationIDParams{})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan estimation was not found. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+		}
+		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan estimation. assetScanEstimationID=%v: %v", assetScanEstimationID, err))
+	}
```

Comment:
No need for this the DB layer will check this for you as part of the Update function